transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java

```diff
@@ -194,7 +193,25 @@ private StructuredRecord createStructuredRecord(CSVRecord record) {
     StructuredRecord.Builder builder = StructuredRecord.builder(outSchema);
     int i = 0;
     for (Field field : fields) {
-      builder.set(field.getName(), TypeConvertor.get(record.get(i), field.getSchema().getType()));
+      String val = record.get(i);
+      Schema fieldSchema = field.getSchema();
+
+      if (val.isEmpty()) {
+        boolean isNullable = fieldSchema.isNullable();
+        Schema.Type fieldType = isNullable ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
+        // if the field is a string or a nullable string, set the value to the empty string
+        if (fieldType == Schema.Type.STRING) {
+          builder.set(field.getName(), "");
+        } else if (!isNullable) {
+          // otherwise, error out
+          throw new IllegalArgumentException(String.format(
+            "The %dth field named '%s' is of non-nullable type '%s', " +
+              "but was parsed as an empty string for CSV record '%s'",
+            i + 1, field.getName(), field.getSchema().getType(), record));
```

Comment:
i like that format better, will change. I put i+1 because "0th" seemed weird, but `Field #0` may be better so there is no confusion if its 0 based or 1 based. Will change that as well.